API/Repositories/Implementations/TournamentsRepository.cs

```diff
@@ -167,7 +170,7 @@ LIMIT @count
 
     private async Task<Tournament> UpdateExisting(TournamentWebSubmissionDTO wrapper)
     {
-        Tournament? existing = await GetAsync(wrapper.TournamentName) ?? throw new Exception("Tournament does not exist, this method assumes the tournament exists.");
+        Tournament? existing = await SearchSingleAsync(wrapper.TournamentName) ?? throw new Exception("Tournament does not exist, this method assumes the tournament exists.");
```

Comment:
You should have another repository function that checks the `ILike` *without* a wildcard (`%`) on each end to indicate case-insensitive title matching. Wildcards don't work here because it leaves room for false matches.

Modifying `SearchSingleAsync` to accept a `bool partialMatch` or similar should be sufficient.

Reply:
I get what you mean but still want to clarify, `ILike` is what enables the case insensitivity vs `Like`, and the `%` wrapper enables the partial matching.

Reply:
Wait nvm that's exactly what you said, I just woke up and haven't had enough coffee 🤦